doc/samtools.1

```diff
@@ -953,7 +953,9 @@ constructed using %\fInum\fRs and %s notation, consuming \fInum\fR
 characters of the MD5sum.  For example
 \fB/local/ref_cache/%2s/%2s/%s\fR will create 2 nested subdirectories
 with the filenames in the deepest directory being the last 28
-characters of the md5sum.
+characters of the md5sum.  If no \fB%s\fR rules are present in the
+REF_CACHE value it will it assumed to end with an implicit \fB%s\fR, ie
```

Comment:
```suggestion
REF_CACHE value it will be assumed to end with an implicit \fB%s\fR, i.e.
```

Also this is not quite true in two ways: there's a bunch of code about adding a `/` if necessary, so it usually gets an implicit `/%s` appended; this happens if there's some `%2s` rules but no all-remaining `%s` rule. So maybe write as “If no without-digits `%s` rules” (or something…) and “using no additional subdirectories” (also no hyphen in subdirectories).

Reply:
Thanks, how about this?

Reply:
The (other) scenario I was thinking would be confusing is when I set `REF_CACHE=foo/bar` then “using no subdirectories” is confusing because it seems to say it'll undo my `bar`. The new addition sadly makes the “ie…” harder to parse…

It's pretty hard for users to know what we're trying to communicate to them with this :smile:

In the context of the whole paragraph, what about a different approach (probably as its own paragraph right after this one)? 

> If you have set up a nested tree structure (e.g. with _misc/seq_cache_populate.pl_), you will need to use …/%2s/… correspondingly in `REF_PATH` and/or `REF_CACHE` so that tools will be able to use your tree structure correctly.

Reply:
Well the "using no subdirectories" does immediately follow the sentence with the %2s/%2s/%s and "2 nested subdirectories", so I was assuming the reader could put 2 and 2 together. :-)

However I'll take another bash at it.

Reply:
Third try.  This needed a much bigger restructuring to avoid using seq_cache_populate in an example before it's described.  I also tried to clarify a few other minor bits.  I kept the bit about an implicit trailing /%s as it does need documenting somewhere.

Reply:
:+1: